### PR TITLE
IPv4 address selection

### PR DESCRIPTION
**Pull request by jukkar**
Select IPv4 source address according to destination address. This is needed in multi network interface setup.

### CODE REVIEW

File: include/net/net_if.h [hunk 1]
```diff
@@ -1362,6 +1362,17 @@ struct net_if_router *net_if_ipv4_router_add(struct net_if *iface,
 bool net_if_ipv4_addr_mask_cmp(struct net_if *iface,
```
Comment (pfl) [inline include/net/net_if.h]: The commit message could better state that the interface selection happens via the assigned subnet as Zephyr does not support generic routes to be set. Then this addition is understandable that it looks at the netmask o fthe interface only.

**Review by pfl — CHANGES_REQUESTED** [review]
Commit message of the first patch.

File: include/net/net_if.h (hunk 1 above)
Comment (jukkar) [reply include/net/net_if.h]: Yes, that makes sense.

**Comment by jukkar** [comment]
Added more text to 1st commit description.

**Review by pfl — APPROVED** [review]

**Review by tbursztyka — APPROVED** [review]

**Pull request merged**